sklearn/preprocessing/_encoders.py

```diff
@@ -571,7 +571,7 @@ def _legacy_fit_transform(self, X):
                                 " 'auto', int or array of ints, got %r"
                                 % type(self._n_values))
             if n_values.ndim < 1 or n_values.shape[0] != X.shape[1]:
-                raise ValueError("Shape mismatch: if n_values is an array,"
+                raise ValueError("Shape mismatch: if categories is an array,"
```

Comment:
I don't think the change is applicable here. This condition applies only when the user is working with the deprecated parameter.

Reply:
will revert the change for the _legacy_fit_transform.